docs/tutorials/tfx/stub_template.ipynb

```diff
@@ -0,0 +1,821 @@
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "colab_type": "text",
+        "id": "ydsPE-aN1Alm"
+      },
+      "source": [
+        "##### Copyright &copy; 2020 The TensorFlow Authors."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "colab_type": "text",
+        "id": "uCSByGH6C7zS"
+      },
+      "source": [
+        "<font size=-1>Licensed under the Apache License, Version 2.0 (the \\\"License\\\");\n",
+        "you may not use this file except in compliance with the License.\n",
+        "You may obtain a copy of the License at [https://www.apache.org/licenses/LICENSE-2.0](https://www.apache.org/licenses/LICENSE-2.0)\n",
+        "\n",
+        "Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an \\\"AS IS\\\" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the License for the specific language governing permissions and limitations under the License.</font>"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "colab_type": "text",
+        "id": "6TyrY7lV0oke"
+      },
+      "source": [
+        "# TFX Template Pipeline Testing Framework using Stub Executors"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "colab_type": "text",
+        "id": "iLYriYe10okf"
+      },
+      "source": [
+        "## Introduction\n",
+        "**You should complete [template.ipynb](https://github.com/tensorflow/tfx/blob/master/docs/tutorials/tfx/template.ipynb) tutorial up to *Step 6* in order to proceed this tutorial.**\n",
+        "\n",
+        "This document will provide instructions to test a TensorFlow Extended (TFX) pipeline\n",
+        "using `BaseStubExecuctor`, which generates fake data using the golden test data. This is intended for users to replace executors they don't want to test so that they could save time from running the actual executors. Stub executor is provided with TFX Python package under `tfx.experimental.pipeline_testing.base_stub_executor`.\n",
+        "\n",
+        "Many of the instructions are Linux shell commands, which will run on an AI Platform Notebooks instance. Corresponding Jupyter Notebook code cells which invoke those commands using `!` are provided.\n",
+        "\n",
+        "This tutorial serves as an extension to `template.ipynb` tutorial, thus you will also use [Taxi Trips dataset](\n",
+        "https://data.cityofchicago.org/Transportation/Taxi-Trips/wrvz-psew)\n",
+        "released by the City of Chicago. We strongly encourage you to try modifying the components prior to utilizing stub executors.\n",
+        "\n"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "colab_type": "text",
+        "id": "jxPMeugQ0okg"
+      },
+      "source": [
+        "## Step 7. Set up your environment.\n",
```

Comment:
I think that we don't have to start this tutorial from Step 7. I think that we can skip all steps for the environment setting and running basic pipelines. Because all of them are already covered in the other tutorials. We might able to try to make this tutorial self-contained, but I'm afraid that it might increase overall maintenance burden.